tools/clang/lib/CodeGen/CGStmt.cpp

```diff
@@ -709,7 +709,12 @@ void CodeGenFunction::EmitCondBrHints(llvm::LLVMContext &Context,
   }
 
   // FIXME: This condition is never false.  Should it be an assert?
-  if (!Metadata.empty()) {
+  if (!Metadata.empty()
+      // HLSL Change Begin.
+      // HLSL attrs will be ignored, then LoopID will get {null} which may shared with other metadata.
+      && Metadata.size() > 1
```

Comment:
Only need to check if Metadata.size() > 1 since that implies its not empty. Add a comment that we only want to enter this if we found a llvm loop attribute and we know we found an llvm attribute if the metadata size > 1.